src/spect_experimental/utilities/ctac_to_mu_values.cxx

```diff
@@ -0,0 +1,224 @@
+
+  while( in_iter != input_image_sptr->end_all_const())
+  {
+    if (*in_iter<0.f) {
+      float mu = a1 + b1 *(*in_iter);
+      *out_iter = (mu < 0.0f) ? 0.0f : mu;
+    } else {
+      *out_iter = a2 + b2 * (*in_iter);
+    }
+
+    ++in_iter; ++out_iter;
+  }
+
+  return Succeeded::yes;
+}
+
+
+int main(int argc, char * argv[])
+{
+  USING_NAMESPACE_STIR;
+  const char * output_filename = 0;
+  const char * input_filename = 0;
+  const char * slope_filename = 0;
+  const char * manufacturer_name = 0;
+  const char * keV_str = 0;
+
+  const char * const usage = "ctac_to_mu_values -o output_filename -i input_dicom_slice -j slope_filename -m manufacturer_name -k target_energy\n";
+  opterr = 0;
+  {
+    char c;
+
+    while ((c = getopt (argc, argv, "i:o:j::m:k:?")) != -1)
+      switch (c)
+	{
+	case 'i':
+	  input_filename = optarg;
+	  break;
+	case 'o':
+	  output_filename = optarg;
+	  break;
+	case 'j':
+      slope_filename = optarg;
+	  break;
+    case 'm':
+      manufacturer_name = optarg;
+      break;
+	case 'k':
+      keV_str = optarg;
+      break;
+	case '?':
+	  std::cerr << usage;
+	  return EXIT_FAILURE;
+	default:
+	  if (isprint (optopt))
+	    fprintf (stderr, "Unknown option `-%c'.\n", optopt);
+	  else
+	    fprintf (stderr,
+		     "Unknown option character `\\x%x'.\n",
+		     optopt);
+	  std::cerr << usage;
+	  return EXIT_FAILURE;
+	}
+  }
+
+  if (output_filename==0 || input_filename==0 || slope_filename==0 || manufacturer_name==0 || keV_str==0 )
+    {
+      std::cerr << usage;
+      return EXIT_FAILURE;
+    }
+
+  //Read slope file
+  std::ifstream slope_json_file_stream(slope_filename);
+  nlohmann::json slope_json;
+  slope_json_file_stream >> slope_json;
+
+  //Put user-specified manufacturer into upper case.
+  std::string manufacturer = manufacturer_name;
+  std::locale loc;
+
+  for (std::string::size_type i=0; i<manufacturer.length(); ++i)
+    manufacturer[i] = std::toupper(manufacturer[i],loc);
+
+  stir::info(boost::format("Manufacturer: '%s'") % manufacturer);
+
+  //Get desired keV as integer value
+  int keV;
+
+  std::stringstream ss;
+  ss << keV_str;
+  ss >> keV;
+
+  stir::info(boost::format("target keV: '%i'") % keV);
+
+  //Extract appropriate chunk of JSON file for given manufacturer.
+  nlohmann::json target = slope_json["scale"][manufacturer]["transform"];
+
+  int location = -1;
+  int pos = 0;
+  for (auto entry : target){
+    if (entry["kev"] == keV)
+      location = pos;
+    pos++;
+  }
+
+  if (location == -1){
+    std::cerr << "Desired keV: " << keV << " not found! ";
+    std::cerr << "Aborting!";
+    return EXIT_FAILURE;
+  }
+
+  //Extract transform for specific keV.
+  nlohmann::json j = target[location];
+  //std::cout << j.dump(4);
+
+  //Read DICOM data
+  stir::info(boost::format("ctac_to_mu_values: opening file %1%") % input_filename);
+  std::unique_ptr< FloatImageType > input_image_sptr(stir::read_from_file<FloatImageType>( input_filename ));
```

Comment:
```suggestion
unique_ptr< FloatImageType > input_image_sptr(stir::read_from_file<FloatImageType>( input_filename ));
```
cannot have `std` due to work-arounds for old compilers

Reply:
Done 0061adf.